### PR TITLE
monster loot charges

### LINKED ISSUE #3881: monster loot charges
### Explanation of what you want to do that is currently impossible

currently if you dont set any charge it will be 1x by default so we should use `<attribute key="charges" value="200" />` in items.xml instead to make the charges max on drop

`monster.loot = {
{id = 2170, chance = 99999}
}`

### PR DESCRIPTION
**Pull request by emil92b**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

**Issues addressed:** 
loot with charges didn't set the charges to max

closes #3881

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: data/scripts/lib/register_monster_type.lua [hunk 1]
```diff
@@ -225,6 +232,10 @@ registerMonsterType.loot = function(mtype, mask)
 					end
 					if children.subType or children.charges then
 						child:setSubType(children.subType or children.charges)
+					else
+						if charges > 0 then
+							child:setSubType(charges)
```
Comment (MillhioreBT) [inline data/scripts/lib/register_monster_type.lua:237]: is it normal to place the default charges from parent to child? shouldn't I get the charges of the itemtype of the child id in question?

Comment (Zbizu) [reply data/scripts/lib/register_monster_type.lua:237]: nah, should make a new variable from `ItemType(child.id):getCharges()`

Comment (emil92b) [reply data/scripts/lib/register_monster_type.lua:237]: correct now?

Comment (Zbizu) [reply data/scripts/lib/register_monster_type.lua:237]: haven't tested, but looks ok

Comment (emil92b) [reply data/scripts/lib/register_monster_type.lua:237]: Untitled

**Review by Zbizu — DISMISSED** [review]

File: data/scripts/lib/register_monster_type.lua [hunk 2]
```diff
@@ -204,6 +204,11 @@ registerMonsterType.loot = function(mtype, mask)
 			end
 			if loot.subType or loot.charges then
```
Comment (nekiro) [inline data/scripts/lib/register_monster_type.lua:205]: define charges as
`local charges = loot.charges or ItemType(loot.id):getCharges()`
then just use charges everywhere where loot.charges is used
charges > 0 check is probably redundant as it wont change anything, unless you want to save that single call

**Pull request merged**